counter/src/main/java/org/infinispan/counter/impl/CounterModuleLifecycle.java

```diff
@@ -90,7 +90,7 @@ private static void registerCounterManager(EmbeddedCacheManager cacheManager, Ba
       EmbeddedCounterManager counterManager = new EmbeddedCounterManager(cacheManager);
       // This must happen before CacheManagerJmxRegistration starts
       registry.registerComponent(CounterManager.class, counterManager, true);
-      if (cacheManager.getCacheManagerConfiguration().jmx().enabled()) {
+      if (SecurityActions.getCacheManagerConfiguration(cacheManager).jmx().enabled()) {
```

Comment:
I would add the `GlobalConfiguration` as a parameter :)

Reply:
change the method signature to `registerCounterManager(EmbeddedCacheManager cacheManager, BasicComponentRegistry registry, GlobalConfiguration configuration)` to avoid creating/invoking the `SecurityActions`

Reply:
Argh I see. I didn't look at the wider code calling `registerCounterManager`.